model/attention.py

```diff
@@ -2,14 +2,23 @@
 import torch.nn as nn
 import torch.nn.functional as F
 
-class AttentionWithSelfAblation(nn.Module):
+from transformer_lens.hook_points import HookPoint, HookedRootModule
+
+class AttentionWithSelfAblation(HookedRootModule):
     def __init__(self, config, layer_id):
         super().__init__()
         self.is_local = (config.attention_layers[layer_id] == "local")
         self.num_heads = config.num_heads
         self.hidden_size = config.hidden_size
         self.head_dim = config.hidden_size // config.num_heads
         self.config = config
+        
+        self.hook_k = HookPoint()
+        self.hook_v = HookPoint()
+        self.hook_q = HookPoint()
+        self.attn_hook = HookPoint()
+        self.context = HookPoint()
```

Comment:
I think all the hooks should have "hook" in the name.

Reply:
I agree, I still find naming variables quite difficult